### ISSUE TITLE
[MySQL] Parsing error with 'algorithm' as column name

### ISSUE THREAD

**Issue by sy6sy2** [sqlfluff/sqlfluff#2639]
Using `algorithm` as column name trigger an unparsable error.

Minimal example to reproduce:

```mysql
CREATE TABLE a (
   algorithm INT
);
```

While this example works fine:

```mysql
CREATE TABLE a (
   `algorithm` INT
);
```

## Expected Behaviour

No unparsable error.

## Observed Behaviour

```
L:   2 | P:   1 |  PRS | Line 2, Position 1: Found unparsable section: 'CREATE TABLE a
                       | (algorithm INT)'
```

## Steps to Reproduce

Run `sqlfluff lint --dialect mysql test.sql` on example given above.

## Dialect

Mysql

## Version

0.10.0

**Comment by tunetheweb** [comment]
According to [the mysql docs](https://dev.mysql.com/doc/refman/8.0/en/keywords.html) `ALGORITHM` is a reserved word and "require special treatment for use as identifiers such as table and column names".

This works and is probably good practice even if mysql currently allows the unquoted version:

```sql
CREATE TABLE a (
   `algorithm` INT
);
```

**Comment by barrywhart** [comment]
@tunetheweb: Should this issue be closed?

**Comment by tunetheweb** [comment]
In my opinion yes, but was waiting for @sy6sy2 to see if they agree or if I've got it wrong. Let's close, and re-open if I got it wrong.

**Comment by sy6sy2** [comment]
Thanks to both of you for your fast answers.
No problem to use the quoted version, I opened the issue just in case you wanted to allow the unquoted version in SQLFluff 😉
